lib/logstash/inputs/gelf.rb

```diff
@@ -157,7 +159,7 @@ def self.from_json_parse(json)
   # legacy_parse uses the LogStash::Json class to deserialize json
   def self.legacy_parse(json)
     o = LogStash::Json.load(json)
-    LogStash::Event.new(o) if o
+    LogStash::Event.new(o)
```

Comment:
I have removed the if here, I don't think its possible its possible that the parser return a nil object.